### PR TITLE
[OwnershipVerifier] Use 'printInContext' in all ownership verification errors.

### PR DESCRIPTION
**Pull request by rxwei**
This patch applies printing instructions in context to all lifetime checking and ownership verification errors, as a follow-up to 4662247 where operand ownership verification errors were changed to print instructions in context. This has made debugging significantly easier for me.

### CODE REVIEW

**Comment by rxwei** [comment]
@swift-ci please smoke test

**Comment by rxwei** [comment]
@swift-ci please smoke test

File: lib/SIL/LinearLifetimeChecker.cpp [hunk 1]
```diff
@@ -205,8 +205,11 @@ void State::initializeConsumingUse(BranchPropagatedUser consumingUser,
   error.handleOverConsume([&] {
     llvm::errs() << "Function: '" << value->getFunction()->getName() << "'\n"
                  << "Found over consume?!\n"
-                 << "Value: " << *value << "User: " << *consumingUser
-                 << "Block: bb" << userBlock->getDebugID() << "\n\n";
+                 << "Value:\n";
+    value->printInContext(llvm::errs());
+    llvm::errs() << *value << "User:\n";
```
Comment (gottesmm) [inline lib/SIL/LinearLifetimeChecker.cpp]: This is fine to me. My only request is can you keep each logical print out on the same line. That is, can you put User on a new line with a new invocation of llvm::errs()?

Comment (rxwei) [reply lib/SIL/LinearLifetimeChecker.cpp]: Oh yes, I've actually already printed `value` on the previous line, so this line should just be
```swift
llvm::errs() << "User:\n";
```

Comment (rxwei) [reply lib/SIL/LinearLifetimeChecker.cpp]: Fixed!

**Comment by rxwei** [comment]
@swift-ci please smoke test

**Comment by rxwei** [comment]
Failing tests:
```console
10:28:53 Failing Tests (6):
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/cond_br_crash.sil
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/over_consume.sil
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/leaks.sil
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/disable_verifier_using_semantic_tag.sil
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/arguments.sil
10:28:53     Swift(linux-x86_64) :: SIL/ownership-verifier/subobject_borrowing.sil
```

I'll fix these when I get a chance.

**Comment by gottesmm** [comment]
Feel free to put in a flag that causes the old behavior so the tests pass.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by CodaFi** [comment]
@rxwei Are you still pursuing this?

**Comment by rxwei** [comment]
> @rxwei Are you still pursuing this?

Yes, I'll land it this weekend.

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**